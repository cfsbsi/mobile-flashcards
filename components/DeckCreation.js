import React, {Component} from 'react'
import {View, Text, TextInput, TouchableOpacity} from 'react-native'
import styled from "styled-components/native/index";

class DeckCreation extends Component {

    constructor(props) {
        super(props);
        this.state = {text: ''};
    }

    render() {
        return (
            <AddDeckStyled>
                <QuestionView>
                    <Question>What is the title</Question>
                    <Question>of your new</Question>
                    <Question>deck?</Question>
                </QuestionView>
                <TextInputStyled onChangeText={(text) => this.setState({text})}
                                 value={this.state.text}></TextInputStyled>
                <TouchableOpacityStyled>
                    <TextButton>SUBMIT</TextButton>
                </TouchableOpacityStyled>
            </AddDeckStyled>
        )
    }

}

const AddDeckStyled = styled.View`
    align-items: center;
`

const QuestionView = styled.View`
    margin-top: 70;
    margin-bottom: 20;
`

const TextInputStyled = styled.TextInput`
    margin-top: 40;
    height: 50;
    width: 300;
    border-color: gray;
    border-width: 2;
    border-radius: 7;
`

const Question = styled.Text`
    font-size: 30;
    font-weight: bold;
    text-align: center;
`

const TextButton = styled.Text`
    font-size: 30;
    text-align: center;
    color: white;
`

const TouchableOpacityStyled = styled.TouchableOpacity`
    background-color: black;
    border-radius: 7;
    margin-top: 80;
    padding: 10px 15px;
`

export default DeckCreation