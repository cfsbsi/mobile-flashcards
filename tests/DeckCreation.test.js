import React from 'react';
import renderer from 'react-test-renderer';
import DeckCreation from "../components/DeckCreation";

it('renders correctly', () => {
    const tree = renderer.create(<DeckCreation />).toJSON();
    expect(tree).toMatchSnapshot();
});