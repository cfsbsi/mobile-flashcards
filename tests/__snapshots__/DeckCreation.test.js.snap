// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-around",
      },
      undefined,
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      What is the title
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      of your new
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      deck?
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    style={
      Array [
        Object {
          "borderColor": "gray",
          "borderRadius": 7,
          "borderWidth": 2,
          "height": 50,
          "width": 300,
        },
        undefined,
      ]
    }
    value=""
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "borderRadius": 7,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 30,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      SUBMIT
    </Text>
  </View>
</View>
`;
